tests/IntegrationTests/RedirectionTests.cs

```diff
@@ -0,0 +1,200 @@
+using System.Globalization;
+using System.Net;
+using System.Net.Sockets;
+
+namespace IntegrationTests;
+
+public class RedirectionTests : IClassFixture<DatabaseFixture>, IDisposable
+{
+	public RedirectionTests(DatabaseFixture database)
+	{
+		m_database = database;
+		m_database.Connection.Open();
+	}
+
+	public void Dispose()
+	{
+		m_database.Connection.Close();
+	}
+
+
+	[Fact]
+	public void RedirectionTest()
+	{
+		StartProxy();
+
+		// wait for proxy to launch
+		Thread.Sleep(50);
+		var csb = AppConfig.CreateConnectionStringBuilder();
+		var initialServer = csb.Server;
+		var initialPort = csb.Port;
+		var permitRedirection = true;
+		try
+		{
+			m_database.Connection.Execute(
+				$"set @@global.redirect_url=\"mariadb://{initialServer}:{initialPort}\"");
```

Comment:
Looks like this fails with MariaDB 11.4 since it does accept this global variable, but doesn't include it in the initial OK packet.

Reply:
Usually I handle server-specific functionality by adding a new enum to `ServerFeatures` and skipping the test based on whether that's set in `config.json`.